### PR TITLE
update k8s registry domain

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area rules

**What this PR does / why we need it**:

- update k8s registry domain

part of https://github.com/kubernetes/k8s.io/issues/4780

/assign @leogr 

Not sure if we will need to keep both for while. will wait for some feedback

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by leogr — COMMENTED** [review]
Since the rules can also run on environments not up-to-date, I'd keep both domains (ie `k8s.grc.io` and `registry.k8s.io`).

@LucaGuerra @darryk10 @loresuso wdyt? :thinking:

**Comment by LucaGuerra** [comment]
@leogr Makes a lot of sense to me. Keeping both domains means that we'll be compatible with both newer and older k8s versions. Thanks a lot @cpanato for adding the new ones!

File: rules/falco_rules.yaml
Comment (cpanato) [inline rules/falco_rules.yaml:1862]: @leogr @LucaGuerra  added this date to we remove the k8s reference after December 
the k8s registry will be frozen on 3rd of April 2023 (https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/) so we give some extra time for people to update the clusters.
wdyt?

**Review by leogr — APPROVED** [review]
LGTM.

/hold for a second opinion @falcosecurity/rules-maintainers

**Comment by LucaGuerra** [comment]
LGTM!

**Review by jasondellaluce — APPROVED** [review]
/approve

/unhold

Thanks!

**Pull request merged**
